Proto-nitrate bz response gas reaction can now emit nuclear particles.

Pickle-Coding opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
The proto-nitrate bz response gas reaction can now emit nuclear particles. One particle can be created per 5 moles of BZ spent in the reaction, up to 12 particles in a single reaction costing 60 moles of BZ. The intensity of the radiation pulses in the reaction have been reduced based on how many nuclear particles have been created. The direction is completely random and out of user control. This has the same requirements as emitting radiation pulses, so putting it in waste or distro won't work.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Adds a fun yet dangerous interaction for atmospheric technicians to play around with. Proto-nitrate tritium response and BZ response felt too similar in what they do, so this should make the reactions feel less duplicated.  While it might be possible to use this in a traitorous act, it shouldn't be a very effective choice as the reaction has extremely tight temperature range requirements, and a relatively high energy release compared to its heat capacity, making it only feasible at high efficiency in a controlled and stationary environment.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Proto-nitrate bz response can now emit nuclear particles.
balance: Proto-nitrate bz response radiation pulse intensity will get reduced based on how many particles it is creating.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pickle-Coding committed:
Nuclear particles.

Allows the proto-nitrate BZ response gas reaction to fire nuclear particles. It can fire up to a maximum of 12 nuclear particles per gas mixture per reaction tick, requiring 5 moles of BZ to be consumed per nuclear particle.

Pickle-Coding committed:
My first commit was cursed.

Fixes cursed for loop. Changes how much radiation it costs to make a nuclear particle, and turns it into a define.

Pickle-Coding committed:
Let people know.

Adds nuclear particle description for proto-nitrate bz response gas reaction factor.

Pickle-Coding committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into fun.

This is my merge conflict resolution attempt, and I hope I don't fuck things up.

LemonInTheDark commented:
no

LemonInTheDark commented on code/modules/atmospherics/gasmixtures/reactions.dm:
```diff
@@ -1152,7 +1152,11 @@
+		var/nuclear_particle_amount = min(round(consumed_amount / PN_BZASE_NUCLEAR_PARTICLE_DIVISOR), PN_BZASE_NUCLEAR_PARTICLE_MAXIMUM)
+		for(var/i in 1 to nuclear_particle_amount)
+			location.fire_nuclear_particle()
+		radiation_pulse(location, max_range = min(sqrt(consumed_amount - nuclear_particle_amount * PN_BZASE_NUCLEAR_PARTICLE_RADIATION_ENERGY_CONVERSION) / PN_BZASE_RAD_RANGE_DIVISOR, GAS_REACTION_MAXIMUM_RADIATION_PULSE_RANGE), threshold = PN_BZASE_RAD_THRESHOLD_BASE * INVERSE(PN_BZASE_RAD_THRESHOLD_BASE + consumed_amount - nuclear_particle_amount * PN_BZASE_NUCLEAR_PARTICLE_RADIATION_ENERGY_CONVERSIO ...
```
Why uh, why is the max reaction rad range 40x40 tiles?

Pickle-Coding replied:
This was the max range of all the radiation pulses from all gas reactions. I turned it into a hard limit define, because anything above that would be unfun for crew.

LemonInTheDark replied:
this is sus but lets deal with it later

LemonInTheDark commented on code/__DEFINES/reactions.dm:
```diff
@@ -259,3 +259,9 @@
+/// A scaling factor for the nuclear particle production generated when proto-nitrate breaks down BZ.
+#define PN_BZASE_NUCLEAR_PARTICLE_DIVISOR 5
+/// The maximum amount of nuclear particles that can be produced from proto-nitrate breaking down BZ.
+#define PN_BZASE_NUCLEAR_PARTICLE_MAXIMUM 12
```
This seems really high, especially since this mix is likely to be on more then one tile at once. Worries me

Pickle-Coding replied:
I have no problem with lowering it. Do you have any suggestions for a more appropriate value that's less worrying? I might go with 6 if that's okay.

Pickle-Coding replied:
I have reduced it to 6 per gasmixture per atmos tick. I can still lower it if it's still too high.

Pickle-Coding committed:
Scary to be less scary.

Reduces the maximum amount of nuclear particles the proto_nitrate_bz_response gas reaction can create in a single tick to 6, from 12, by reducing the PN_BZASE_NUCLEAR_PARTICLE_MAXIMUM define to 6, from 12. 12 particles per gasmixture per tick is a bit scary for both performance and gameplay reasons, 6 may be less scary, for both performance and gameplay reasons.
